### PR TITLE
feature: move to github action #242

### PR DESCRIPTION
**Pull request by Bablzz**
Hello everyone!

In this MR I add github action. There are the same steps as in .travis.yml

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Go
+
+on: [push, pull_request]
+
+jobs:
+  shellcheck:
```
Comment (mdelapenya) [inline .github/workflows/ci.yml:6]: wdyt of renaming the entire job to this, which is more inclusive with other tasks than shellcheck?
```suggestion
  static-analysis:
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: Go
```
Comment (mdelapenya) [inline .github/workflows/ci.yml:1]:
```suggestion
name: Main pipeline
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        run: go mod tidy
+
+      - name: vet
+        run: go vet ./...
+
+      - name: gotestsum
+        run: gotestsum --format short-verbose ./...
+
+      - name: Test
+        run: go test -v ./...
```
Comment (mdelapenya) [inline .github/workflows/ci.yml:53]: I think this stage is already executed by the previous one
```suggestion

```

**Review by mdelapenya — COMMENTED** [review]
Hi @Bablzz, thank you for this great contribution. I added a few nits about names, but on the other hand it will LGTM after hearing your opinion.

Thanks! 👋

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Bablzz) [reply .github/workflows/ci.yml:53]: Agree

**Review by mdelapenya — APPROVED** [review]
LGTM 👍 

image

**Comment by mdelapenya** [comment]
@gianarb I think we can keep travis for a while, comparing executions, and once we are happy with GH, then move on removing Travis descriptor.

**Comment by gianarb** [comment]
THanks @Bablzz !!!

**Comment by Bablzz** [comment]
@gianarb @mdelapenya My pleasure! :handshake:

**Pull request merged**